R/utils.R

```diff
@@ -18,3 +18,30 @@ tick <- function(x) {
 default_local_src <- function() {
   src_df(env = .GlobalEnv)
 }
+
+
+clean_up_dm <- function(dm, tables, gotta_rename) {
+  start <- tables[1]
+  # filters need to be empty, for the disambiguation to work
+  # the renaming will be minimized, if we reduce the `dm` to the necessary tables here
+  red_dm <-
+    cdm_reset_all_filters(dm) %>% cdm_select_tbl(tables)
+
+  if (gotta_rename) {
+    recipe <-
+      compute_disambiguate_cols_recipe(red_dm, tables, sep = ".")
+    explain_col_rename(recipe)
+    # prepare `dm` by disambiguating columns (on a reduced dm)
+    clean_dm <-
+      col_rename(red_dm, recipe)
+    # the column names of start_tbl need to be updated, since taken from `dm` and not `clean_dm`,
+    # therefore we need a named variable containing the new and old names
+    renames <-
+      recipe %>% filter(table == !!start) %>% pull() %>% flatten_chr()
```

Comment:
Not sure which is better:

```suggestion
      recipe$...[recipe$table == "start"][[1]]
```

Reply:
After trying out some things the following seemed to be at least not completely without grace...
```r
pluck(recipe[recipe$table == start, "renames"], 1, 1)
```

Reply:
`[` is similar to the pipe because you can read left to right. Why not:

```r
recipe$renames[recipe$table == "start"][[1]][[1]]
```

(Do we need the second `[[1]]` ?)

Reply:
Ok, this works (actually without the second `[[1]]`, otherwise the names are lost)

Reply:
just realized that it doesn't work for empty lists (when no renaming is necessary), since `list()[[1]]` throws an error.

Reply:
what about `pluck(recipe$renames[recipe$table == start], 1)`? (works for empty list)